internal/database/dbutil/db_type.go

```diff
@@ -35,54 +36,266 @@ func GetDbTypeFrom(backend types.BackendType, valueType types.ValueType) (string
 }
 
+func ValueType(backend types.BackendType, dbValueType string) (types.ValueType, error) {
+	var mp map[string]types.ValueType
+	switch backend {
+	case types.POSTGRES:
+		mp = postgresTypeToValueType
+	case types.MYSQL:
+		mp = mySQLTypeToValueType
+	case types.SNOWFLAKE:
+		mp = snowflakeTypeToValueType
+	case types.BIGQUERY:
+		mp = bigQueryTypeToValueType
+	case types.SQLite:
+		mp = sqliteTypeToValueType
+	case types.REDSHIFT:
+		mp = redshiftTypeToValueType
+	default:
+		return 0, errdefs.InvalidAttribute(fmt.Errorf("unsupported backend: %s", backend))
+	}
+
+	t, ok := mp[strings.ToLower(dbValueType)]
+	if !ok {
+		return 0, errdefs.InvalidAttribute(fmt.Errorf("unsupported db value type: %s", dbValueType))
+	}
+	return t, nil
+}
+
+// Mapping feature value type to database data type
 var (
-	sqliteType = map[types.ValueType]string{
-		types.STRING:  "TEXT",
-		types.INT64:   "INTEGER",
-		types.FLOAT64: "FLOAT",
-		types.BOOL:    "INTEGER",
-		types.BYTES:   "BLOB",
-		types.TIME:    "TIMESTAMP",
-	}
-	mysqlType = map[types.ValueType]string{
+	valueTypeToSQLiteType = map[types.ValueType]string{
+		types.STRING:  "text",
+		types.INT64:   "integer",
+		types.FLOAT64: "float",
+		types.BOOL:    "integer",
+		types.BYTES:   "blob",
+		types.TIME:    "timestamp",
+	}
+	valueTypeToMySQLType = map[types.ValueType]string{
 		types.STRING:  "text",
 		types.INT64:   "bigint",
 		types.FLOAT64: "double",
 		types.BOOL:    "bool",
 		types.TIME:    "datetime",
-		types.BYTES:   "binary",
+		types.BYTES:   "varbinary",
 	}
-	postgresType = map[types.ValueType]string{
+	valueTypeToPostgresType = map[types.ValueType]string{
 		types.STRING:  "text",
 		types.INT64:   "bigint",
 		types.FLOAT64: "double precision",
 		types.BOOL:    "boolean",
-		types.TIME:    "timestamptz",
+		types.TIME:    "timestamp",
 		types.BYTES:   "bytea",
 	}
-	snowFlake = map[types.ValueType]string{
-		types.STRING:  "VARCHAR",
-		types.INT64:   "INTEGER",
-		types.FLOAT64: "DOUBLE",
-		types.BOOL:    "BOOLEAN",
-		types.TIME:    "TIME",
-		types.BYTES:   "BINARY",
+	valueTypeToSnowFlake = map[types.ValueType]string{
+		types.STRING:  "varchar",
+		types.INT64:   "bigint",
+		types.FLOAT64: "double",
+		types.BOOL:    "boolean",
+		types.TIME:    "timestamp",
+		types.BYTES:   "varbinary",
 	}
-	dynamoDB = map[types.ValueType]string{
+	valueTypeToDynamoDB = map[types.ValueType]string{
 		types.STRING:  "String",
 		types.INT64:   "BigInteger",
-		types.FLOAT64: "Float",
+		types.FLOAT64: "Double",
 		types.BOOL:    "Boolean",
 		types.TIME:    "Date",
-		types.BYTES:   "Byte",
+		types.BYTES:   "ByteBuffer",
 	}
-	cassandraType = map[types.ValueType]string{
+	valueTypeToCassandraType = map[types.ValueType]string{
 		types.STRING:  "text",
 		types.INT64:   "bigint",
 		types.FLOAT64: "double",
 		types.BOOL:    "boolean",
 		types.TIME:    "timestamp",
-		types.BYTES:   "text",
+		types.BYTES:   "blob",
+	}
+	valueTypeToRedshiftType = valueTypeToPostgresType
```

Comment:
this may not work since redshift doesn't have type `bytea` according to https://docs.aws.amazon.com/redshift/latest/dg/c_Supported_data_types.html

Reply:
This line was copied from master:

https://github.com/oom-ai/oomstore/blob/f6390b31e2847e9f8dfee86d4ce7a548ab3f8ce9/internal/database/dbutil/db_type.go#L87

Reply:
then my previous code doesn't work 😅 let's fix it here

Reply:
ok

Reply:
I create an issue to track this problem. https://github.com/oom-ai/oomstore/issues/782